File: 1291. sequential-digits.py
from typing import List

class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:

        digits = "123456789"

        ans = []

        min_len = len(str(low))
        max_len = len(str(high))

        for length in range(min_len, max_len + 1):

            for start in range(10 - length):

                num = int(digits[start:start + length])

                if low <= num <= high:
                    ans.append(num)

        return ans


if __name__ == "__main__":
    sol = Solution()

    print(sol.sequentialDigits(
        100, 300
    ))  # Expected: [123, 234]

    print(sol.sequentialDigits(
        1000, 13000
    ))  # Expected: [1234, 2345, 3456, 4567, 5678, 6789, 12345]

File: 1291. sequential-digits.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> sequentialDigits(int low, int high) {

        String digits = "123456789";

        List<Integer> ans = new ArrayList<>();

        int minLen = String.valueOf(low).length();
        int maxLen = String.valueOf(high).length();

        for (int len = minLen; len <= maxLen; len++) {

            for (int start = 0; start + len <= 9; start++) {

                int num = Integer.parseInt(digits.substring(start, start + len));

                if (num >= low && num <= high) {
                    ans.add(num);
                }
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println(sol.sequentialDigits(
            100, 300
        )); // Expected: [123, 234]

        System.out.println(sol.sequentialDigits(
            1000, 13000
        )); // Expected: [1234, 2345, 3456, 4567, 5678, 6789, 12345]
    }
}